### PR TITLE
Fix service id not initialized

### LINKED ISSUE #744: Search crash
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.

Unfortunately I can't reproduce this crash, as far as I remember I had searched for something and was going through the list and watched videos and went back to the search result and at some point I think I double clicked back and it crashed but I can't reproduce it. I thought it was worth posting the stacktrace just in case:

[newpipev010exception3.txt](https://github.com/TeamNewPipe/NewPipe/files/1336998/newpipev010exception3.txt)

### PR DESCRIPTION
**Pull request by coffeemakr**
- [x] I carefully reed the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

Code improvements and proguard rules for icepick (copy pasted from the [README](https://github.com/frankiesardo/icepick)).

This should fix all serviceId == -1 issues and maybe some others. The problem is that this only occurs when using the release configuration because Proguard probably removed some icepick annotation.  

This closes #744  and #749

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/util/ExtractorHelper.java [hunk 1]
```diff
@@ -50,7 +50,14 @@ private ExtractorHelper() {
         //no instance
     }
 
+    private static void checkServiceId(int serviceId) {
```
Comment (mauriciocolli) [inline app/src/main/java/org/schabi/newpipe/util/ExtractorHelper.java]: IMO we don't need to check that. All the functions uses `NewPipe.getService`, which throws a self-explanatory error in the case of an non-existent service: "There's no service with the id = xx".

Comment (coffeemakr) [reply app/src/main/java/org/schabi/newpipe/util/ExtractorHelper.java]: Yes the functions use this method but deferred. The exception may thrown at any time in any thread and the calling method is not shown. 
Have a look at https://sentry.schabi.org/sentry/newpipe/issues/382/. There is no simple way to see from which fragment the serviceId is provided.

Comment (coffeemakr) [reply app/src/main/java/org/schabi/newpipe/util/ExtractorHelper.java]: So it this a blocking issue?

**Comment by coffeemakr** [comment]
I don't mean to be rude but this bug should be fixed ASAP. Generate a bugfix release or whatever but this issue has 10 reports a day.

**Comment by theScrabi** [comment]
@coffeemakr Yea kind of true. So the way I see this is that we don't know where this issue happens, right?

@mauriciocolli If it helps we might need to introduce that checkServiceId.
In my opinion it would not hurt to have this function.

**Comment by coffeemakr** [comment]
* `checkServiceId` is to fail faster.
 * It's clear that **all saved states are not restored** because IcePick is used and Proguard removes it.

**Pull request merged**